lib/classes/console.js

```diff
@@ -0,0 +1,256 @@
+    this.stateSchemaVersion = '1';
+    this.extensionLayerName = 'sls-console-otel-extension';
+  }
+
+  async initialize() {
+    this.isEnabled = (() => {
+      const {
+        configurationInput: configuration,
+        processedInput: { commands, options },
+      } = this.serverless;
+      if (!_.get(configuration, 'console')) return false;
+      this.org = options.org || configuration.org;
+      if (!this.org) return false;
+      const command = commands.join(' ');
+      if (!supportedCommands.has(command)) return false;
+
+      const providerName = configuration.provider.name || configuration.provider;
+      if (providerName !== 'aws') {
+        log.error(`Provider "${providerName}" is currently not supported by the console`);
+        return false;
+      }
+
+      if (command !== 'rollback' && (command !== 'deploy' || !options.package)) {
+        if (
+          !Object.values(this.serverless.service.functions).some((functionConfig) =>
+            this.isFunctionSupported(functionConfig)
+          )
+        ) {
+          log.warning(
+            "Cannot enable console: Service doesn't configure any function with supported runtime"
+          );
+          return false;
+        }
+      }
+      return true;
+    })();
+    if (!this.isEnabled) return;
+    if (!isAuthenticated()) {
+      const errorMessage = process.env.CI
+        ? 'You are not currently logged in. Follow instructions in http://slss.io/run-in-cicd to setup env vars for authentication.'
+        : 'You are not currently logged in. To log in, run "serverless login"';
+      throw new ServerlessError(errorMessage, 'CONSOLE_NOT_AUTHENTICATED');
+    }
+    this.provider = this.serverless.getProvider('aws');
+    this.sdk = await getPlatformClientWithAccessKey(this.org);
+    this.orgId = (await this.sdk.getOrgByName(this.org)).orgUid;
+    this.service = this.serverless.service.service;
+    this.stage = this.provider.getStage();
+    this.otelIngestionUrl = (() => {
+      if (process.env.SLS_CONSOLE_OTEL_INGESTION_URL) {
+        return process.env.SLS_CONSOLE_OTEL_INGESTION_URL;
+      }
+      return process.env.SERVERLESS_PLATFORM_STAGE === 'dev'
+        ? 'https://core.serverless-dev.com/ingestion/kinesis'
+        : 'https://core.serverless.com/ingestion/kinesis';
+    })();
+  }
+
+  isFunctionSupported({ handler, runtime }) {
+    if (!handler) return false; // Docker container image (not supported yet)
+    if (!runtime) return true; // Default is supported nodejs runtime
+    return runtime.startsWith('nodejs');
+  }
+
+  async createOtelIngestionToken() {
+    const url = `${this.otelIngestionUrl}/org/${this.orgId}/service/${this.service}/stage/${this.stage}`;
+    const data = {
+      method: 'GET',
+      headers: {
+        'Authorization': `Bearer ${this.sdk.accessKey}`,
+        'Content-Type': 'application/json',
+      },
+    };
+    log.debug('get token: %s %o', url, data);
+    const response = await fetch(url, data);
```

Comment:
IS there a reason why we're doing it this way instead of having this method implemented on SDK level?

Reply:
That's a good point. I'll discuss in later turn a future of SDK, and if we agree we want to stick to it I'll move it there

Reply:
sounds good 👍